### ISSUE TITLE
 jest-styled-components uses global.beforeEach rather than global.beforeAll or both?

### ISSUE THREAD

**Issue by JeromeGill** [styled-components/jest-styled-components#400]
This is probably more a question than an issue...

 jest-styled-components uses [global.beforeEach](https://github.com/JeromeGill/jest-styled-components/blob/main/src/index.js#L5) to reset stylesheets which means in situations where you don't want to remount a component for every test case for whatever reason, you cannot use beforeAll to load it.
 
 Is there a reason jest-styled-components needs to reset stylesheets in beforeEach? If so, is there a reason _not_ to fire it in beforeAll as well?

**Comment by ruslanguns** [comment]
I believe this should be at the developer's convenience, but I think it should be a beforeAll( ) or via an opt-in option.

**Comment by d3vl1m3** [comment]
Yep. Just got caught out by this too. Any resolution for this? Would be a good to have this option to use `beforeAll`.
